runtime_scan/pkg/provider/aws/client.go

```diff
@@ -69,97 +69,28 @@ func (c Client) Kind() models.CloudProvider {
 	return models.AWS
 }
 
-func (c *Client) DiscoverScopes(ctx context.Context) (*models.Scopes, error) {
-	regions, err := c.ListAllRegions(ctx, true)
-	if err != nil {
-		return nil, fmt.Errorf("failed to list all regions: %w", err)
-	}
-
-	scopes := models.ScopeType{}
-	err = scopes.FromAwsAccountScope(models.AwsAccountScope{
-		Regions: convertToAPIRegions(regions),
-	})
-	if err != nil {
-		return nil, FatalError{
-			Err: fmt.Errorf("failed to cast AwsAccountScope to ScopeType: %w", err),
-		}
-	}
-
-	return &models.Scopes{
-		ScopeInfo: &scopes,
-	}, nil
-}
-
 // nolint:cyclop
-func (c *Client) DiscoverTargets(ctx context.Context, scanScope *models.ScanScopeType) ([]models.TargetType, error) {
-	var filters []ec2types.Filter
-
-	awsScanScope, err := scanScope.AsAwsScanScope()
+func (c *Client) DiscoverTargets(ctx context.Context) ([]models.TargetType, error) {
+	regions, err := c.ListAllRegions(ctx)
 	if err != nil {
-		return nil, FatalError{
-			Err: fmt.Errorf("failed to cast ScanScopeType to AwsAccountScope: %w", err),
-		}
+		return nil, fmt.Errorf("failed to get regions: %w", err)
```

Comment:
not sure I can skip this error, DescribeRegions is a single API call which gets all the regions that are available from AWS. We could definitely change the logic below so that we process all the regions one after the other along with the channel idea from above that way we can show the error at the end.

Reply:
sorry, the comment in a wrong place. Right, I meant to try to do it best effort and not return on getVMInfoFromInstance(), GetInstances()